src/core/document.js

```diff
@@ -574,40 +575,12 @@ class PDFDocument {
   parse(recoveryMode) {
     this.setup(recoveryMode);
```

Comment:
Now that you have *significantly* simplified the `parse` method, and thank you for doing that since it always felt a bit unwieldy, I believe that you can simply inline `PDFDocument.setup` here given that it only contains two lines of code:
https://github.com/mozilla/pdf.js/blob/34cec29288212a6ca5bb9d703d6c5102c47e635d/src/core/document.js#L706-L709